### ISSUE TITLE
consider writing specprod_dir to headers

### ISSUE THREAD

**Issue by moustakas** [desihub/desispec#1077]
As far as I can tell, `SPECPROD_DIR` is not written to any of the headers of files generated by the pipeline (especially `cframe` and `coadd`), but as an end-user, it would be very useful to have it accessible this way (i.e., without setting the environment variable). Could we add this header card?

**Comment by sbailey** [comment]
Added DEPNAMnn/DEPVERnn keywords for both $DESI_SPECTRO_REDUX and $SPECPROD if they are defined; use `desiutil.depend.getdep()` to parse these.  This is admittedly slightly less convenient than storing SPECPROD_DIR=$DESI_SPECTRO_REDUX/$SPECPROD, but slightly less ambiguous if the code is run outside of the context of a production when those variables might not be set.  Reopen to ask again if you really want them combined.

**Comment by moustakas** [comment]
I'm likely missing something, but I was hoping I could glean the production name `denali` from `SPECPROD`, but the dependency in the individual cframe files is cut off and in any case it's not propagated to the coadds.

Let me know if you'd like me to try to push a fix.

```
cd /global/cfs/cdirs/desi/spectro/redux/denali/exposures/20210224
fitsheader 00077980/cframe-b5-00077980.fits  | grep SPECPROD
  DEPNAM02= 'SPECPROD'
  IN_PSF  = 'SPECPROD/exposures/20210224/00077980/psf-b5-00077980.fits' / Input sp
  IN_IMG  = 'SPECPROD/preproc/20210224/00077980/preproc-b5-00077980.fits'
  ORIG_PSF= 'SPECPROD/calibnight/20210224/psfnight-b5-20210224.fits'
  IN_FRAME= 'SPECPROD/exposures/20210224/00077980/frame-b5-00077980.fits'
  FIBERFLT= 'SPECPROD/calibnight/20210224/fiberflatnight-b5-20210224.fits'
  IN_SKY  = 'SPECPROD/exposures/20210224/00077980/sky-b5-00077980.fits'
  IN_CALIB= 'SPECPROD/exposures/20210224/00077980/fluxcalib-b5-00077980.fits'
```

Missing from coadds:

```
cd /global/cfs/cdirs/desi/spectro/redux/denali/tiles/cumulative/80607/20201219
fitsheader coadd-1-80607-thru20201219.fits | grep SPECPROD
fitsheader zbest-1-80607-thru20201219.fits | grep SPECPROD
```

**Comment by moustakas** [comment]
`SPECPROD` is still missing in Everest, whether as a `DEPNAMnn/DEPVERnn` keyword or something else. I'm specifically looking at the `coadd-`, `spectra-` and `redrock-` files.

**Comment by moustakas** [comment]
Thanks for https://github.com/desihub/desiutil/pull/183, @sbailey. 

Is there a header I can inspect? In particular, I'm hoping that we're also writing the equivalent of `coadd_type` to the header, too (`per-exposure`, `per-night`, `cumulative`, `healpix`) but it wasn't clear from that PR. Otherwise, this information still has to be inferred from the filename and/or directory structure where the file is found.

**Comment by sbailey** [comment]
coadd type isn't there yet, but that's coming before launching fuji...

And unfortunately I've been re-using the same prod for testing N>>1 PRs today so I've already blown away the test files for this, but I'll make a clean "state of affairs" run tonight with all the latest changes for inspection tomorrow.

**Comment by sbailey** [comment]
addressed in PR #1618.  Closing.
